src/libexpr/eval-settings.hh

```diff
@@ -19,18 +19,18 @@ struct EvalSettings : Config
         Enable built-in functions that allow executing native code.
 
         In particular, this adds:
-        - `builtins.importNative` *path*
-        
-          Load a dynamic shared object (DSO) at *path* which exposes a function pointer to a procedure that initialises a Nix language value, and return that value.
+        - `builtins.importNative` *path* *symbol*
+          Runs function with *symbol* from a dynamic shared object (DSO) at *path*.
+          This may be used to add new primitive functions to the Nix language, that may not be possible to express within the language itself.
```

Comment:
Not sure if I understand the first statement correctly. Does `symbol` represent a Nix language value that is passed as `Value & v`? 

```suggestion
          Run a procedure from a dynamic shared object (DSO) at *path* on a Nix value *symbol* .
```

The second sentence is misleading because it vaguely suggests this may be the only or even a remotely good way to do things. Arguably, *anything* can be expressed in the Nix language itself, and there are other ways to add primitives. I'll of course merge concrete, self-contained examples that illustrate how the primitive works and what to consider when using it, but for now this short description is as accurate as it gets.

Reply:
@Mic92 that was actually a question to you: what do you mean exactly? The new wording "loads function with the symbol" does not clarify what's actually happening. What is that symbol? Is it in the DSO or in the Nix language?

Reply:
Both. First it's a nix string value by the user and will be passed as a string argument to dlopen().

Reply:
```
The function dlsym() takes a "handle" of a dynamic library returned by dlopen() and the null-terminated symbol name, returning the address where that symbol is loaded into memory.
```

Reply:
@fricklerhandwerk do you have a better way to express that, otherwise I will just close this pull request, also the current documentation is incorrect.

Reply:
Ah yes, makes sense to me now.